src/components/projects-carousel/style.css

```diff
@@ -0,0 +1,77 @@
+.carousel {
+  position: relative;
+  container-name: projects-carousel;
+  container-type: inline-size;
+}
+
+.carousel-images {
+  overflow: hidden;
+  gap: 2rem;
+}
+
+.slide {
+  width: 99%;
+  height: 99%;
+  margin-inline: auto;
+  padding: 2.75rem 1.125rem 3.375rem 1.875rem;
+  outline: 2px solid var(--color-secondary-light);
+  border-radius: 1rem;
+  @container projects-carousel (min-width: 40em) {
+    padding: 3.5rem 6.5rem 4rem;
+  }
+  .icon-circle {
+    flex-shrink: 0;
```

Comment:
Flex shrink 0 pode trazer alguns problemas, como overflow. Eu sugiro estruturar esses elementos um pouco diferente:
1) colocar width e height de 100% no svg;
2) colocar width no icon-circle. Eu iria com um valor em porcentagem dentro de um clamp, eu testei clamp(50px, 12%, 80px) e curti.
3) Colocar flex-wrap no flex container. Aqui eh uma questão que é bom discutir com design. O flex wrap vai impedir que o flex container encolha os elementos filhos. Se eles nao quiserem is por esse caminho, eu aumentaria o flex-shrink do h3, em vez de zerar o flex shrink do icone. 

 Essa é uma sugestão de caminho, faça os testes aí e vê o que acha =)

Reply:
não consegui chegar num resultado que agradasse com o clamp no width, o botão variava um bocado de formato, não sei se fiz algo errado, talvez só com o flex-wrap no pai já resolva e não tenhamos risco de overflow já que estamos trabalhando os tamanhos e espaços com container queries, faz sentido?

mas se tivesses um tempinho queria entender como ficaria isso que propusesses

Reply:
O formato estranho eh pq o svg tem width e height chumbados em 46. Se colocar width e height 100% resolve. 
No icon-circle eu botei width: clamp(50px, 12%, 80px);
    aspect-ratio: 1;
   E tirei o flex-shrink 0. 
   
   Mas de novo, nao precisa implementar isso assim nao. So presta atencao no tamanho do icone e do icon-ricle em telas menores. Eu to achando um pouco sem respiro/padding